hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/Checksum.java

```diff
@@ -60,7 +61,7 @@ private static Function<ByteBuffer, ByteString> newMessageDigestFunction(
   }
 
   public static ByteString int2ByteString(int n) {
-    return ByteString.copyFrom(Ints.toByteArray(n));
+    return UnsafeByteOperations.unsafeWrap(Ints.toByteArray(n));
```

Comment:
Should this be conditional on the following config property?

https://github.com/apache/ozone/blob/df9ed54ba1235044fee73a7709b376b9dd44b213/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java#L102-L105

Similar to current logic for `ByteBuffer`s:

https://github.com/apache/ozone/blob/df9ed54ba1235044fee73a7709b376b9dd44b213/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ByteStringConversion.java#L45-L53

Reply:
Yeah..i thought about that but
(1) it seems quite involved to pass the configuration object over to make the decision
(2) while it's called "unsafe byte operations", because the  byte array passed in is guaranteed immutable (Ints.toByteArray(n) returns a new temporary byte array), it is safe.

Reply:
Makes sense, thanks.